src/components/popover/popover.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+    Component,
+    h,
+    Prop,
+    Element,
+    Event,
+    EventEmitter,
+} from '@stencil/core';
+import { createRandomString } from '../../util/random-string';
+
+/**
+ * @slot - header - Content to put inside the header surface
+ * @slot - body - Content to put inside the body surface
+ * @slot - footer - Content to put inside the footer surface
+ * @exampleComponent limel-example-popover
+ */
+
+@Component({
+    tag: 'limel-popover',
+    shadow: true,
+})
+export class Popover {
+    /**
+     * True if the content within the popover should be visible
+     */
+    @Prop()
+    public visible = false;
+
+    /**
+     * Emits an event when the component is closing
+     */
+    @Event()
+    private close: EventEmitter<void>;
+
+    @Element()
+    private host: HTMLLimelPopoverElement;
+
+    private portalId: string;
+
+    constructor() {
+        this.portalId = createRandomString();
+        this.globalClickListener = this.globalClickListener.bind(this);
+    }
+
+    public componentDidLoad() {
+        document.addEventListener('click', this.globalClickListener, {
+            passive: true,
+        });
+    }
+
+    public disconnectedCallback() {
+        document.removeEventListener('click', this.globalClickListener);
+    }
+
+    public render() {
+        return (
+            <limel-portal visible={this.visible} containerId={this.portalId}>
+                <limel-popover-surface
+                    open={this.visible}
+                    contentCollection={this.host.children}
+                >
+                    <slot name="header" />
+                    <slot name="body" />
+                    <slot name="footer" />
+                </limel-popover-surface>
+            </limel-portal>
+        );
+    }
+
+    private globalClickListener({ target }) {
```

Comment:
How about making this a bit more explicit? What do you think about
```suggestion
    private globalClickListener(event: MouseEvent) {
```